### PR TITLE
WIP: compare machine files to make automatical changes

### PR DESCRIPTION
**Pull request by MichelleXiaoxiaoWang**

### CODE REVIEW

File: Jenkins_jobs/CreateNewNode.groovy [hunk 1]
```diff
@@ -16,20 +16,88 @@ node {
...
+                }
+                }
+                def name = "";
+                if (!EXISTED){
+                    machine.find (/\S+.ibm.com/){
+                        def location = it.indexOf(".")
+                        it = it.substring(0,location)
+                        name = it
+                    }
+                    machines = machines + "${name}"
```
Comment (VermaSh) [inline Jenkins_jobs/CreateNewNode.groovy]: Can you please add comments here with an sample example and maybe a one line explanation

File: Jenkins_jobs/CreateNewNode.groovy [hunk 2]
```diff
@@ -70,7 +142,7 @@ node {
                    (Jenkins.getInstance().getComputer(newMachineName)).connect(false);
                 }
 
-                println "Machine ${newMachineName} was added with following labels ${newMachineLabels}";
+                println "Machine:'${newMachineName}'\nlabels: '${newMachineLabels}'\nremote root directory: '${remoteFS}'";
```
Comment (VermaSh) [inline Jenkins_jobs/CreateNewNode.groovy]: Maybe add separators before and after the print statement to make it easier to separate different machines apart in the output logs

File: vars/Constants.groovy [hunk 3]
```diff
@@ -1,18 +1,16 @@
 class Constants {
+    static final String USERNAME = "j9build";
```
Comment (VermaSh) [inline vars/Constants.groovy]: We shouldn't be checking in internal values to the outside world

File: vars/Constants.groovy [hunk 4]
```diff
@@ -1,18 +1,16 @@
...
     static final String[] IGNORE_LABELS = ['.ignore'];
-    static final String REMOTE_FS = "/home/jenkins";
-    static final String WIN_REMOTE_FS = "C:\\Users\\jenkins";
+    static final String REMOTE_FS = "/home/${USERNAME}";
+    static final String WIN_REMOTE_FS = "C:\\Users\\${USERNAME}";
+    static final String OSX_REMOTE_FS = "/Users/${USERNAME}";
     
     // This the key that'll be used for SSHLauncher in CreateNewNode
-    static final String SSH_CREDENTIAL_ID = "";
+    static final String SSH_CREDENTIAL_ID = "83181e25-eea4-4f55-8b3e-e79615733226";
```
Comment (VermaSh) [inline vars/Constants.groovy]: Same here

File: Jenkins_jobs/CreateNewNode.groovy [hunk 5]
```diff
@@ -16,20 +16,88 @@ node {
...
         String[] machineIPs = params.machineIPs.split(",")
         String[] labels = params.labelStrings.split(",")
+        def FILES = []
+        git url: "git@github.ibm.com:Xiaoxiao-Wang/infra-can.git", credentialsId: Constants.SSH_CREDENTIAL_ID, branch:"INI"
+        FILES = sh (
+                    script: "cat ansible/playbooks/Infrastructure/dynamic_inventory.ini",
+                    returnStdout: true
+                ).trim()
+        def FILES_LIST = FILES.split("\\n")
+        echo "${FILES_LIST}"
```
[the remark this answers is not in the document]
Comment (MichelleXiaoxiaoWang) [reply Jenkins_jobs/CreateNewNode.groovy]: fixed

File: Jenkins_jobs/CreateNewNode.groovy [hunk 6]
```diff
@@ -16,20 +16,88 @@ node {
...
+        for (machine in FILES_LIST){ 
+            def EXISTED = false;
+            def emptyMap = [:];
```
[the remark this answers is not in the document]
Comment (MichelleXiaoxiaoWang) [reply Jenkins_jobs/CreateNewNode.groovy]: fixed

File: Jenkins_jobs/CreateNewNode.groovy (hunk 1 above)
Comment (MichelleXiaoxiaoWang) [reply Jenkins_jobs/CreateNewNode.groovy]: commented

File: Jenkins_jobs/CreateNewNode.groovy [hunk 7]
```diff
@@ -16,20 +16,88 @@ node {
...
+            def halfmachine = "";
+            if (machine =~ /[^\s]+ \s*(\S+)\s*=\s*(.*)?(?:#|$)\S*/) {
+            if (machine.contains("#")) {
+                def location = machine.indexOf("#")
+                halfmachine = machine.substring(location + 1,machine.length())
+            } else {
+                halfmachine = ""
+            }
+            def splitLines = halfmachine.split(',')
```
[the remark this answers is not in the document]
Comment (MichelleXiaoxiaoWang) [reply Jenkins_jobs/CreateNewNode.groovy]: fixed

File: Jenkins_jobs/CreateNewNode.groovy [hunk 8]
```diff
@@ -16,20 +16,88 @@ node {
...
+        FILES = sh (
+                    script: "cat ansible/playbooks/Infrastructure/dynamic_inventory.ini",
+                    returnStdout: true
+                ).trim()
+        def FILES_LIST = FILES.split("\\n")
+        echo "${FILES_LIST}"
+        for (machine in FILES_LIST){ 
+            def EXISTED = false;
+            def emptyMap = [:];
+            def halfmachine = "";
```
[the remark this answers is not in the document]
Comment (MichelleXiaoxiaoWang) [reply Jenkins_jobs/CreateNewNode.groovy]: fixed

**Comment by karianna** [comment]
There's also some merge conflicts now so you may have to rebase.

**Comment by VermaSh** [comment]
@MichelleXiaoxiaoWang is the regular expression only supposed to pick up lines with # at the end?

**Comment by karianna** [comment]
@MichelleXiaoxiaoWang this needs a rebase

**Comment by sxa** [comment]
@MichelleXiaoxiaoWang @VermaSh This PR is marked as WIP and has had no activity in nearly four years - I assume it is no longer required and can therefore be closed?

**Comment by AdamBrousseau** [comment]
I think this is fine to close.

**Pull request closed without merging**